server/context.go

```diff
@@ -95,6 +95,12 @@ func (s *SessionManager) nextPid() uint64 {
 	return s.pid
 }
 
+func (s *SessionManager) InitialSession(conn *mysql.Conn) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.sessions[conn.ConnectionID] = &managedSession{nil, conn}
```

Comment:
This is definitely a little wonky. Open to suggestions here.

What we're dealing with: `*Handler.NewConnection` gets called pretty much as soon as the connection is accepted and assigned a unique connection ID. At that point, the connection handshake is ongoing and there is no authentication information, TLS is still to be negotiated, there is no initial schema selected, etc. After all of that is done, `*Handler.ComInitDB` is called, at which point the session is fully established. (`ComInitDB` will be called even with `schemaName == ""` in our fork of Vitess).

SessionManager is where our implementation of `KillConnection(connID uint32)` lives. If we want connections which are currently authenticating to be killable with `*SessionManager.KillConnection`, we need to have them registered with SessionManager in NewConnection and not only after ComInitDB.

An alternative approach would be to put these into a different map. SessionManager could have a `sessionsByConnID map[uint32]sql.Session` and `connections map[uint32]*mysql.Conn`. This method could be called `AddConn` instead, it would just insert into `connections`. And `CloseConn` could then remove from both.

Reply:
I went ahead and did this. Feel free to push back if I made it worse :).